app/controllers/application_controller.rb

```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
   include SessionsHelper
+  before_action :set_global_search_variable
+
+  def search
```

Comment:
should not be in application_controller but there should be a separate controller for this

Reply:
please discuss